src/Evolution/Systems/ScalarWave/Characteristics.cpp

```diff
@@ -0,0 +1,171 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/ScalarWave/Characteristics.hpp"
+
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Variables.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace ScalarWave {
+template <size_t Dim>
+void characteristic_speeds(
+    const gsl::not_null<std::array<DataVector, 4>*> char_speeds,
+    const tnsr::i<DataVector, Dim,
+                  Frame::Inertial>& /*unit_normal_one_form*/) noexcept {
+  (*char_speeds)[0] = 0.;   // v(UPsi)
+  (*char_speeds)[1] = 0.;   // v(UZero)
+  (*char_speeds)[2] = 1.;   // v(UPlus)
+  (*char_speeds)[3] = -1.;  // v(UMinus)
+}
+
+template <size_t Dim>
+std::array<DataVector, 4> characteristic_speeds(
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        unit_normal_one_form) noexcept {
+  auto char_speeds = make_with_value<std::array<DataVector, 4>>(
+      get<0>(unit_normal_one_form), 0.);
```

Comment:
Why initialize with `make_with_value` when you will overwrite the value?

Does something like `auto char_speeds = make_array<4>(DataVector(get<0>(unit_normal_one_form).size()));` work?

Reply:
Giving this a little more thought, I don't think my suggestion works... keep what you had :)

Reply:
:)